### PR TITLE
Add black and flake8

### PR DESCRIPTION
**Pull request by fornwall**
This PR adds:

- Code formatting using [black](https://black.readthedocs.io/en/stable/). This tool seems to be gaining a lot of momentum, has been moved into the Python Software Foundation, and is a nice way to minimise diffs and not having to care about code formatting (just make it auto-format using black on save).
- Style guide checks using [flake8](https://flake8.pycqa.org/en/latest/).
- Makes `python setup.py test` run everything (black, flake8, pytest, pyright) so you don't have to think about the individual steps (and it simplifies the README somewhat).

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[flake8]
```
Comment (fornwall) [inline .flake8:1]: This .flake8 configuration file is the one from https://github.com/psf/black, which makes flake8 not complain about code formatted by black.

Comment (ksaaskil) [reply .flake8:1]: Could you add these also to `meeshkan-codestyle` repo so it's consistent across repos? Thanks!

Comment (fornwall) [reply .flake8:1]: 👍Adding in https://github.com/Meeshkan/meeshkan-codestyle/pull/2

**Review by ksaaskil — APPROVED** [review]
:+1:

**Pull request merged**